source/pod.go

```diff
@@ -114,7 +115,16 @@ func (ps *podSource) Endpoints(ctx context.Context) ([]*endpoint.Endpoint, error
 			domainList := annotations.SplitHostnameAnnotation(domainAnnotation)
 			for _, domain := range domainList {
 				if len(targets) == 0 {
-					node, _ := ps.nodeInformer.Lister().Get(pod.Spec.NodeName)
+					node, err := ps.nodeInformer.Lister().Get(pod.Spec.NodeName)
+					if err != nil {
+						if errors.IsNotFound(err) {
```

Comment:
are we sure that it is a IsNotFound ? Could you point me to docs. Maybe we could have just a blanket 
```
if err != nil {
  break or continue
}
```